### PR TITLE
zephyr-xenlib integration to the app

### PR DESCRIPTION
**Pull request by oleksiimoisieiev**
Integrating zephyr-xenlib to the application. 
Includes Xen domain support, Xrun support and Littlefs support.

This branch requires changes to west.yaml in order to set to the latest zephyr-xenlib:
`
remotes:
    - name: amoi
      url-base: https://github.com/oleksiimoisieiev/
projects:
     - name: zephyr-xenlib
      remote: amoi
      revision: xrun_upstr
`

### CODE REVIEW

File: prj.conf [hunk 1]
```diff
@@ -23,6 +23,37 @@ CONFIG_PTHREAD_IPC=y
...
+CONFIG_JSON_LIBRARY=y
+
 CONFIG_XEN_SHELL=y
+CONFIG_XRUN_SHELL_CMDS=y
+# Set mem parameters
+
+CONFIG_MAX_XLAT_TABLES=24
+CONFIG_HEAP_MEM_POOL_SIZE=2097152
+
+CONFIG_USERSPACE=y
```
Comment (al1img) [inline prj.conf:40]: Is it mem paramter?

File: prj.conf [hunk 2]
```diff
@@ -23,6 +23,37 @@ CONFIG_PTHREAD_IPC=y
...
+
+CONFIG_SHELL=y
```
Comment (al1img) [inline prj.conf:42]: It is not mem parameters.

File: prj.conf [hunk 3]
```diff
@@ -23,6 +23,37 @@ CONFIG_PTHREAD_IPC=y
...
+CONFIG_SHELL_BACKEND_SERIAL=y
+
+CONFIG_DEBUG_INFO=y
```
Comment (al1img) [inline prj.conf:45]: Do we need debug info?

File: prj.conf [hunk 4]
```diff
@@ -23,6 +23,37 @@ CONFIG_PTHREAD_IPC=y
...
+
+CONFIG_OVERRIDE_FRAME_POINTER_DEFAULT=y
+CONFIG_OMIT_FRAME_POINTER=n
+
+CONFIG_MAIN_STACK_SIZE=32768
```
Comment (al1img) [inline prj.conf:50]: Not bad. I like when there is some pattern :)

File: CMakeLists.txt [hunk 5]
```diff
@@ -11,6 +11,16 @@ find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 
 include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
 
+set(UTILS_SCRIPTS_DIR ${CMAKE_CURRENT_SOURCE_DIR}/scripts)
```
Comment (al1img) [inline CMakeLists.txt:14]: Move inside if. Use small latter for local variables.

Comment (oleksiimoisieiev) [reply CMakeLists.txt]: +

File: prj.conf (hunk 2 above)
Comment (oleksiimoisieiev) [reply prj.conf:53]: +

File: prj.conf (hunk 3 above)
Comment (oleksiimoisieiev) [reply prj.conf]: +

File: west.yml [hunk 6]
```diff
@@ -18,6 +18,9 @@ manifest:
     - name: aoscloud
       url-base: https://github.com/aoscloud
 
+    - name: zephyr
```
[the remark this answers is not in the document]
Comment (oleksiimoisieiev) [reply west.yml:21]: sorry but it's not scope of this MR

File: west.yml [hunk 7]
```diff
@@ -27,3 +27,7 @@ manifest:
     - name: aos_core_lib_cpp
       remote: aoscloud
       revision: develop
+
```
[the remark this answers is not in the document]
Comment (oleksiimoisieiev) [reply west.yml:33]: +

File: CMakeLists.txt [hunk 8]
```diff
@@ -84,6 +84,19 @@ ExternalProject_Add(
...
+        add_custom_target(
+            fix_dtb
+            DEPENDS ${ZEPHYR_FINAL_EXECUTABLE}
+            COMMAND ${utils_scripts_dir}/fix_dtb.sh ${utils_scripts_dir}/gicv2.tmpl ${ZEPHYR_BINARY_DIR}/virt_gicv2.dtb
+                    ${ZEPHYR_BINARY_DIR}/zephyr.bin
+        )
+    else()
+        message(WARNING "No dtc program")
+    endif()
+endif()
```
[the remark this answers is not in the document]
Comment (oleksiimoisieiev) [reply CMakeLists.txt:115]: +

**Review by Eabramov84 — APPROVED** [review]

**Pull request merged**